File: 2025_03/leetcode2025-03-14.py
class Solution:
    def isBalanced(self, num: str) -> bool:
        n = len(num)
        odd, even = 0, 0
        for i in range(n):
            if i&1:
                odd += int(num[i])
            else:
                even += int(num[i])
        return odd==even

if __name__ == "__main__":
    s = Solution()
    num = "1234"
    print(s.isBalanced(num))

File: 2025_03/leetcode2025-03-14.cpp
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    bool isBalanced(string num) {
        int odd = 0, even = 0;
        int n = num.size();
        for (int i=0; i<n; i++){
            if (i&1){
                odd += (num[i]-'0');
            }
            else{
                even += (num[i]-'0');
            }
        }
        return odd == even;
    }
};

int main(){
    Solution s;
    string num = "1234";
    cout << s.isBalanced(num) << endl;
}
